src/Microsoft.ML.Data/Data/SchemaDefinition.cs

```diff
@@ -17,50 +17,65 @@ namespace Microsoft.ML.Data
     [AttributeUsage(AttributeTargets.Field | AttributeTargets.Property, AllowMultiple = false, Inherited = true)]
     public sealed class KeyTypeAttribute : Attribute
     {
-        // REVIEW: Property based, but should I just have a constructor?
+        /// <summary>
+        /// Specifies expected size of key type.
+        /// </summary>
+        /// <param name="count">Size of key type.</param>
+        public KeyTypeAttribute(ulong count)
```

Comment:
>KeyTypeAttribute [](start = 15, length = 16)

Can this one mirror our existing constructors for the key type itself, that is, parameterless constructor where the result winds up being a key type with the maximum applicable count for the type specified?

Reply:
So basically no size here, and in place where we consume this count, just look on member type and use maxvalue for that type.
I can do that.
But there is small issue with MF I think.
It actually rely on key size and have loops based on size of key.
Which can blew up any MF calculation.

@wschin is it possible to change MF code to not rely on key size?

---
In reply to: [258276031](https://github.com/dotnet/machinelearning/pull/2638#discussion_r258276031) [](ancestors = 258276031)

Reply:
I think it's fine to keep specifying the key size. Just so long as we have another constructor where it is not specified.

---
In reply to: [258277629](https://github.com/dotnet/machinelearning/pull/2638#discussion_r258277629) [](ancestors = 258277629,258276031)

Reply:
So, I think it's fine for MF to fail with an informative error message if it an input is somehow judged as "too big."